extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/filters/accesslog/DefaultAccessLogReceiver.java

```diff
@@ -201,111 +157,68 @@ public void run() {
+                    log.error(e);
                 }
             }
         }
     }
 
-    /**
-     * For tests only. Blocks the current thread until all messages are written
-     * Just does a busy wait.
-     * <p/>
-     * DO NOT USE THIS OUTSIDE OF A TEST
-     */
-    void awaitWrittenForTest() throws InterruptedException {
-        while (!pendingMessages.isEmpty() || forceLogRotation) {
-            Thread.sleep(10);
+    private void writeMessage(final List<String> messages) {
+        if (now().toEpochMilli() > changeOverPoint) {
+            rotateToNextLogFile();
         }
-        while (state != 0) {
-            Thread.sleep(10);
+        try {
+            messages.forEach(this::writeMessage);
+            writer.flush();
+        } catch (IOException e) {
+            log.error(e);
         }
     }
 
-    private void writeMessage(final List<String> messages) {
-        if (System.currentTimeMillis() > changeOverPoint) {
-            doRotate();
-        }
+    private void writeMessage(String message) {
         try {
-            if (writer == null) {
-                boolean created = !Files.exists(defaultLogFile);
-                writer = Files.newBufferedWriter(defaultLogFile, StandardCharsets.UTF_8, StandardOpenOption.APPEND,
-                        StandardOpenOption.CREATE);
-                if (Files.size(defaultLogFile) == 0 && fileHeaderGenerator != null) {
-                    String header = fileHeaderGenerator.generateHeader();
-                    if (header != null) {
-                        writer.write(header);
-                        writer.write("\n");
-                        writer.flush();
-                    }
-                }
-            }
-            for (String message : messages) {
-                writer.write(message);
-                writer.write('\n');
-            }
-            writer.flush();
+            writer.write(message);
+            writer.newLine();
         } catch (IOException e) {
-            log.error("Error writing access log", e);
+            log.error(e);
         }
     }
 
-    private void doRotate() {
-        forceLogRotation = false;
+    private void rotateToNextLogFile() {
         if (!rotate) {
             return;
         }
         try {
-            if (writer != null) {
-                writer.flush();
-                writer.close();
-                writer = null;
-            }
+            writer.close();
             if (!Files.exists(defaultLogFile)) {
                 return;
             }
-            Path newFile = outputDirectory.resolve(logBaseName + currentDateString + logNameSuffix);
-            int count = 0;
-            while (Files.exists(newFile)) {
-                ++count;
-                newFile = outputDirectory.resolve(logBaseName + currentDateString + "-" + count + logNameSuffix);
-            }
-            Files.move(defaultLogFile, newFile);
+            Files.move(defaultLogFile,
```

Comment:
its copied away to having increment. so having one writer on the first file seems doable. flushing before file roation.